libdnf/module/modulemd/ModuleMetadata.cpp

```diff
@@ -193,4 +193,32 @@ void ModuleMetadata::reportFailures(const GPtrArray *failures)
     }
 }
 
+ModulemdObsoletes * ModuleMetadata::getNewestActiveObsolete(ModulePackage *p)
+{
+    ModulemdModule * myModule = modulemd_module_index_get_module(resultingModuleIndex, p->getName().c_str());
```

Comment:
Minor comments:
I am not happy when single letter variable is used. See libsolv and p and pp variables in macros and both have type int.
Also call p->getName().c_str() is not much efficient. Internally  it calls `modulemd_module_stream_get_module_name(mdStream)`. When it returns null it is converted into empty string. and empty string is not NULL.